modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/resolver/CollectionValueResolver.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.resolver;
+
+import static org.mule.module.extension.internal.util.IntrospectionUtils.checkInstantiable;
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.lifecycle.Lifecycle;
+import org.mule.module.extension.internal.util.MuleExtensionUtils;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A {@link ValueResolver} that takes a list of {@link ValueResolver}s
+ * and upon invocation of {@link #resolve(MuleEvent)} it return a
+ * {@link Collection} of values with the outcome of each original resolver.
+ * <p/>
+ * This class implements {@link Lifecycle} and propagates those events to each
+ * of the {@code resolvers}
+ *
+ * @param <T>
+ * @since 3.7.0
+ */
+public final class CollectionValueResolver<T> implements ValueResolver<Collection<T>>
```

Comment:
Yes and no. This is a composite in the sense that it contains more Valueresolvers, but it's not in the sense that what I'd expect from a CompositeValueResolver is to still keep a signature which accepts any return type T. 

This on the other hand, has a more limited semantics, since it's for only returning a Collection<T> ... I think that the current name is more intention revealing since it makes it clear that the output is a collection, while CompositeValueResolver should have a broader contract.